fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -1649,23 +1658,47 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 			writeThread = createGenericThreadPool();
 			readThreads = createGenericThreadPool();
 		}
-		struct Writer* writer =
-		    new Writer(db, defaultFdbCF, id, readIterPool, perfContextMetrics, SERVER_KNOBS->ROCKSDB_READ_PARALLELISM);
 		if (SERVER_KNOBS->ROCKSDB_HISTOGRAMS_SAMPLE_RATE > 0) {
-			actors.push_back(updateHistogram(writer->metricPromiseStream.getFuture()));
+			collection = actorCollection(addActor.getFuture());
 		}
-		writeThread->addThread(writer, "fdb-rocksdb-wr");
-		TraceEvent("RocksDBReadThreads").detail("KnobRocksDBReadParallelism", SERVER_KNOBS->ROCKSDB_READ_PARALLELISM);
-		for (unsigned i = 0; i < SERVER_KNOBS->ROCKSDB_READ_PARALLELISM; ++i) {
-			struct Reader* reader = new Reader(db, defaultFdbCF, readIterPool, perfContextMetrics, i);
+		for (int i = 0; i < SERVER_KNOBS->ROCKSDB_READ_PARALLELISM + 1; i++) {
+			// ROCKSDB_READ_PARALLELISM readers and 1 writer
+			metricPromiseStreams.emplace_back(
+			    std::make_unique<ThreadReturnPromiseStream<std::pair<std::string, double>>>());
 			if (SERVER_KNOBS->ROCKSDB_HISTOGRAMS_SAMPLE_RATE > 0) {
-				actors.push_back(updateHistogram(reader->metricPromiseStream.getFuture()));
+				addActor.send(updateHistogram(metricPromiseStreams[i]->getFuture()));
 			}
-			readThreads->addThread(reader, "fdb-rocksdb-re");
+		}
+		if (SERVER_KNOBS->ROCKSDB_HISTOGRAMS_SAMPLE_RATE > 0) {
+			actorErrorListener = errorListenActor(collection);
+		}
+		writeThread->addThread(new Writer(db,
+		                                  defaultFdbCF,
+		                                  id,
+		                                  readIterPool,
+		                                  perfContextMetrics,
+		                                  SERVER_KNOBS->ROCKSDB_READ_PARALLELISM,
+		                                  metricPromiseStreams[SERVER_KNOBS->ROCKSDB_READ_PARALLELISM].get()),
```

Comment:
nit: add comments here about why `ROCKSDB_READ_PARALLELISM` is the index of a writer metric. This is a little hard to read w/o context.